File: backend/internal/testdata/sample.c
// Sample C fixture simulating CPython object model

#define Py_INCREF(op) (((PyObject*)(op))->ob_refcnt++)

typedef struct _object {
    int ob_refcnt;
    void *ob_type;
} PyObject;

typedef struct _typeobject {
    const char *tp_name;
    int tp_basicsize;
} PyTypeObject;

PyTypeObject PyLong_Type = {
    "int",
    sizeof(PyObject)
};

static PyObject* _PyLong_New(long v) {
    PyObject *obj = 0;
    return obj;
}

PyObject* PyLong_FromLong(long v) {
    PyTypeObject *tp = &PyLong_Type;
    PyObject *result = _PyLong_New(v);
    Py_INCREF(result);
    return result;
}

File: backend/internal/testdata/sample.py
# Sample Python fixture simulating CPython high-level bindings

DEFAULT_FLAGS = 0

class IntObject:
    def __init__(self, value: int):
        self.value = value

    def to_string(self) -> str:
        return str(self.value)

def create_int(val: int) -> IntObject:
    obj = IntObject(val)
    return obj

def process_numbers():
    first = create_int(42)
    s = first.to_string()
    return s

